internal/mode/static/manager.go

```diff
@@ -110,13 +113,26 @@ func StartManager(cfg config.Config) error {
 		int32(cfg.HealthConfig.Port):  "HealthPort",
 	}
 
+	mustExtractGVK := func(obj client.Object) schema.GroupVersionKind {
+		gvk, err := apiutil.GVKForObject(obj, scheme)
+		if err != nil {
+			panic(fmt.Sprintf("could not extract GVK for object: %T", obj))
+		}
```

Comment:
a unit test needs to be added for these lines

Reply:
That's expected. We can't easily test this function.